### PR TITLE
Extend Timetable to hold 10pm-11pm blocks

### PR DESCRIPTION
**Pull request by russelltankaimin**

### CODE REVIEW

File: src/main/java/seedu/address/model/time/util/TimeUtil.java [hunk 1]
```diff
@@ -19,6 +19,10 @@
  * Contains utils functions for Timetable related jobs.
  */
 public class TimeUtil {
+
+    public static final Integer FIRST_HOUR = 0;
```
Comment (mynameizzhafeez) [inline src/main/java/seedu/address/model/time/util/TimeUtil.java:23]: `First Valid Start Hr`

File: src/main/java/seedu/address/model/time/TimePeriod.java [hunk 2]
```diff
@@ -22,8 +22,8 @@ public abstract class TimePeriod {
      * Constructs a period in time.
      */
     public TimePeriod(LocalTime startTime, LocalTime endTime, Day schoolDay) {
-        if (startTime.isAfter(endTime)) {
-            throw new WrongTimeException("Start Time Cannot be after End Time!");
+        if (startTime.isAfter(endTime) || startTime.isEqual(endTime)) {
```
Comment (mynameizzhafeez) [inline src/main/java/seedu/address/model/time/TimePeriod.java:25]: Alright. Noted. Though a simpler implementation would just be:
```suggestion
        if (endTime.isAfter(startTime)) {
```

**Review by mynameizzhafeez — APPROVED** [review]
LGTM. When you’re ready, just merge.

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 3]
```diff
@@ -240,18 +246,42 @@ public static Day parseDay(String dayAsStr) throws ParseException {
     /**
      * Parses a local time and checks whether it is valid.
      */
-    public static LocalTime parseLocalTime(String localTimeAsStr) throws ParseException {
+    public static LocalTime parseLocalTime(String localTimeAsStr, boolean isStartHour) throws ParseException {
         try {
             int hour = Integer.parseInt(localTimeAsStr);
-            if (!Arrays.asList(Timetable.START_TIMINGS).contains(hour)) {
-                throw new ParseException("Invalid time");
+            if (!TimeUtil.isValidHour(hour)) {
```
Comment (mynameizzhafeez) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:252]: Alright, this is the most ideal way to handle it!

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 4]
```diff
@@ -240,18 +246,42 @@ public static Day parseDay(String dayAsStr) throws ParseException {
...
         } catch (NumberFormatException nfe) {
             throw new ParseException("Invalid time");
         }
     }
 
+    /**
+     * Parses a start time.
+     */
+    public static LocalTime parseStartHour(int hour) throws ParseException {
+        if (TimeUtil.isValidStartHour(hour)) {
```
Comment (mynameizzhafeez) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:267]: This part could use a little bit of happy pathing. The else block could be removed altogether. Or, rather, the throw should be the first thing you do in the if statement, then the actual return statement can be unindented.

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 5]
```diff
@@ -240,18 +246,42 @@ public static Day parseDay(String dayAsStr) throws ParseException {
...
+    }
+
+    /**
+     * Parses an end time.
+     */
+    public static LocalTime parseEndHour(int hour) throws ParseException {
+        if (TimeUtil.isValidEndHour(hour)) {
+            return new LocalTime(hour, 0);
+        } else {
+            throw new ParseException(MESSAGE_INVALID_END_HOUR);
```
Comment (mynameizzhafeez) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:281]: Here as well.

File: src/main/java/seedu/address/model/time/util/TimeUtil.java [hunk 6]
```diff
@@ -19,6 +19,37 @@
...
+     */
+    public static boolean isValidStartHour(int hour) {
+        assert(isValidHour(hour));
+        return hour >= FIRST_START_HOUR && hour <= LAST_START_HOUR;
+    }
+
+    /**
+     * Checks if the time is a valid end hour.
+     */
+    public static boolean isValidEndHour(int hour) {
```
Comment (mynameizzhafeez) [inline src/main/java/seedu/address/model/time/util/TimeUtil.java:48]: Alright, I’ll start using these methods for further code.

**Review by mynameizzhafeez — APPROVED** [review]
LGTM.

File: src/main/java/seedu/address/model/time/TimePeriod.java (hunk 2 above)
Comment (russelltankaimin) [reply src/main/java/seedu/address/model/time/TimePeriod.java:25]: You forgot to negate the predicate

**Pull request merged**